content/blog/egress-traffic-with-kgateway-and-Istio-integration.md

```diff
@@ -0,0 +1,291 @@
+7. Deploy the Ollama Container at port number 11434, binding to 0.0.0.0 so the Kubernetes virtual machine can access it via the host's bridge network.
+   ```
+   docker run -d -v ollama:/root/.ollama -p 11434:11434 -e OLLAMA_HOST=0.0.0.0 ollama/ollama --name ollama-server
+   ```
+8. Get the Container IP of ollama which will be inserted at all the **`address filds` which is 172.17.0.2 in our case.
+   ```
+   docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' ollama-server
+   ```
+9. Deploy the Kyverno Authz Server is a GRPC server capable of processing Envoy External Authorization requests.
+```
+helm install kyverno-authz-server --namespace default --create-namespace --wait \
+  --version 0.1.0 --repo https://kyverno.github.io/kyverno-envoy-plugin \
+  kyverno-authz-server
+```
+
+
+While Istio ambient provides Authorization, Authentication and Egress still, there are several scenarios where kgateway can offer a more powerful alternative:
+
+## Securely Egress Traffic with kGateway + Istio Integration
+
+To establish a dedicated, policy-enforced egress path, we must combine three core resources: the **Istio ServiceEntry** (to register the external host), the **kGateway Egress Gateway** (to serve as the L7 egress waypoint), and the **HTTPRoute** (to apply the routing and security logic).
+
+* Our target is an external Ollama container running on the host machine (host.docker.internal) on the default port 11434 and `ServiceEntries` injects the external Ollama endpoint into the Istio service registry for which we'll use static resolution for our local Docker Desktop bridge.
+* Define the Gateway resource, leveraging the kgateway GatewayClass to instantiate a dedicated proxy that is explicitly listening for outbound traffic to the Ollama host with an HTTPRoute.
+
+```yaml
+kubectl apply -f - <<EOF
+# ollama-serviceentry.yaml - Using DNS for Reliability
+apiVersion: networking.istio.io/v1beta1
+kind: ServiceEntry
+metadata:
+  name: ollama-external-host
+  namespace: default
+  labels:
+    security.corp/egress-approved: "true"
+spec:
+  hosts:
+  - "host.docker.internal"
+  location: MESH_EXTERNAL
+  ports:
+  - number: 11434
+    name: http-ollama
+    protocol: HTTP
+  resolution: DNS
+---
+
+# egress-kgateway.yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: egress-kgateway
+  namespace: default
+spec:
+  gatewayClassName: kgateway
+  listeners:
+    - protocol: HTTP
+      port: 8080
+      name: http
+      allowedRoutes:
+        namespaces:
+          from: All
+---
+
+# ollama-egress-route.yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: ollama-egress-route
+  namespace: default 
+spec:
+  parentRefs:
+  - name: egress-kgateway
+    sectionName: http
+  hostnames:
+  - "host.docker.internal"
+  rules:
+  - backendRefs:
+    - name: ollama-external-host
+      port: 11434
+      kind: ServiceEntry
+      group: "networking.istio.io"
+EOF
```

Comment:
I think it would be good to show that you can route to ollama through the egress gateway here. you can then later show how you limit access to ollama with the authz policies

Reply:
I realized we're also missing the step to add kgateway to the Istio mesh and make sure `curl-test-client` is also in the mesh